### DISCUSSION TITLE
Editor extension for .svx files (Code suggestions, highlighting, formatting)

### DISCUSSION THREAD

**Discussion by codevogel** [pngwn/MDsveX#593] [Q&A]
I love the concept of MDsveX. I'm excited to start using it.

I've created a little playground for myself to test some things out, but quickly ran into issues getting my editor (VSCode) to work nicely with `.svx` files.

The [Svelte for VSCode](https://marketplace.visualstudio.com/items?itemName=svelte.svelte-vscode) extension has brilliant editor functionalities for .svelte files. Great code highlighting, code suggestions, and the [prettier-plugin-svelte](https://github.com/sveltejs/prettier-plugin-svelte) makes for consistent styling. 

This functionality all falls away when I start trying to create `.svx` files..
The current 'best' workaround I found is associating `.svx` as `svelte` files or as `markdown` files. Both of these have major drawbacks. 

Associating with svelte loses markdown highlighting.
![image](https://github.com/pngwn/MDsveX/assets/8700780/f717ddcb-4093-49f5-bb1f-3d042c779edc)

Associating with markdown features good highlighting but no code suggestions.
![image](https://github.com/pngwn/MDsveX/assets/8700780/95463f4c-5355-48aa-b26b-9369c388d6e3)

As there is currently no prettier parser that supports `.svx` files, it also means that you have to lose Prettier formatting altogether, as it will annihilate the Markdown markup in the process of formatting the file if you do attempt to run it.

This lack of editor support really puts me off into picking up MDsvex for my next project. Is there anything I've missed to make life easier, or is this the current state?

**Comment by codevogel** [comment]
I have found this [related issue](https://github.com/pngwn/MDsveX/issues/123) but I'm mostly checking in to see whether there is a better workaround than I'm currently using.

**Comment by lsdebianchi** [comment]
I am in the same exact situation.
I thought that there would have been more IDE support seeing that SVX is offered as an option when creating a new SvelteKit project (congratulation on that btw, it's how I discovered this library).

My current workaround is to treat svx files as svelte (with some word wrap):
```js
// .vscode/settings.json
{
  "files.associations": {
    "*.svx": "svelte"
  },
  "[svelte]": {
    "editor.wordWrap": "on"
  }
}
```

and simply disable the formatting:
```js
// .prettierignore
*.svx
```

So I keep the svelte plugin functionalities and highlights, but I loose markdown highlighting and file formatting. This is the best compromise for now. Still a bit frustrating. It's always crazy how highlight + formatting become important once you got use to them.

It looks like that the [SVX](https://marketplace.visualstudio.com/items?itemName=sebsojeda.vscode-svx) extension only offer syntax highlighting.

**Comment by lure23** [comment]
Interested in the same for IntelliJ IDE's. 

Currently just assigning `.svx` for Markdown plugin.

**Comment by codevogel** [comment]
So... 1.5 years later.

Looking like there's still no LSP support or working syntax highlighting for MDSVex.
Kind of a shame. I've been keeping an eye out for this project, because I really want to use it, but it's hard to take it seriously with the lack of editor support. I almost find it weird that it comes as an option in the `sv` cli, given that there's no proper tooling available. The markdown association workarounds are lacking.

All I can find is [this](https://github.com/pngwn/MDsveX/issues/123#issuecomment-675105816) comment by @pngw from 2020. Are there any actual plans or is there anything on the near horizon?

**Reply by pngwn** [reply]
Hello. 

Your frustrations are totally understandable, I had to take a break from my OSS work because of reasons a little while ago but I've been back and working on stuff this year. 

The tooling issues have been held up by the 1.0 rewrite, which I am happy to say is well under way now after many pauses. When that is done everything else will follow. 

The plan is basically for a big 1.0 release with everything in place, I won't consider it 1.0 until
The following is ready and working:

- a new faster, more resilient parser with typescript support. 
- a new fast, lightweight, modern syntax highlighter. 
- an mdsvex LSP with full svelte/ typescript support. 
- a prettier formatter
- a textmate grammar for syntax highlighting 
- a vscode extension wrapping most of the above (similar to the svelte one)
- new website with better docs, especially around integrating mdsvex with different meta frameworks. 

I don't have a hard timeline but my target is to have the technical work done around the end of the year with a launch early in the new year after everything is ready and has been tested. 

I do work full time so this could change a bit but I think this is a realistic plan.

**Reply by codevogel** [reply]
@pngwn 
Just to clarify, my comments are not meant as complaints or frustration. I genuinely appreciate the work you are doing. That is actually why I am so eager to see the tooling improve: I really want to use mdsvex in real projects, but it is hard to fully commit without proper editor support.

I also completely understand that this is open source work and you are not being paid for it, and that life and full-time work come first. The roadmap you outlined sounds great, and it's good to hear that things are moving forward again.

If it would help, maybe the community could get involved more directly. I for one would be more than happy to contribute however I can to help move things forward.

**Reply by pngwn** [reply]
What I would love, while I'm working on the core parser, is to understand what people want from the tooling and docs. 

I understand that at the least: ts support, a formatter, syntax highlighting. 

For docs: basics, sveltekit, Astro? more?

If someone wanted to create a new issue for tooling and one for new docs, and then start listing out the requirements (and linking any old related issues to the new issues) then that would be amazing. 

Then if people have any experience with the various bits of tooling or would like to write docs/ guides then they could maybe pick stuff up (when it makes sense, the rewrite blocks some stuff).

**Reply by stephenlrandall** [reply]
@pngwn Integration docs for augmenting mdsvex are really important to me -- how I can write plugins to replace unified ones, in particular.

I'm also curious -- are these:

> - an mdsvex LSP with full svelte/ typescript support.
> ...
> - a textmate grammar for syntax highlighting
> - a vscode extension wrapping most of the above (similar to the svelte one)

all done/nearing completion? If you're looking for community support, I'd love to help out with any of these. I've written a personal VS Code extension for mdsvex-like work at my company with basic features (syntax highlighting, code folding, outline pane integration, etc.) but stopped short of full Svelte intellisense integration.

For the past few months, I've been debating dropping mdsvex in favor of a unified-based system I own entirely, but it does just feel like I'm reinventing mdsvex at that point. My issue is wanting Svelte 5 support in combination with custom components (#738), and feeling frustrated that I'm stuck in a weird limbo here. If there's anything I can do to help with mdsvex 1.0, please let me know.

**Reply by xxieen** [reply]
Any update on this?
